tests/integration/rag/test_reranker.py

```diff
@@ -0,0 +1,358 @@
+import os
+
+import pytest
+from deepeval import evaluate
+from deepeval.metrics import ContextualPrecisionMetric, GEval
+from deepeval.test_case import LLMTestCase, LLMTestCaseParams
+from langchain_core.documents import Document
+
+from rag.reranker.reranker import (
+    LLMReranker,
+    document_to_str,
+    format_documents,
+    format_queries,
+)
+from utils.models.factory import ModelType
+
+
+@pytest.fixture
+def semantic_similarity_metric(evaluator_model):
+    return GEval(
+        name="Semantic Similarity",
+        evaluation_steps=[
+            "Evaluate whether two answers are semantically similar or convey the same meaning.",
+            "Lightly penalize omissions of detail, focusing on the main idea.",
+            "Vague language is permissible.",
+        ],
+        evaluation_params=[
+            LLMTestCaseParams.ACTUAL_OUTPUT,
+            LLMTestCaseParams.EXPECTED_OUTPUT,
+        ],
+        model=evaluator_model,
+        # TODO: Increase the threshold.
```

Comment:
How is this a TODO and when to we do this?

Reply:
Updated.

Reply:
Thanks!